email-address-parser/email-addresses.mjs

```diff
@@ -0,0 +1,1087 @@
+
+    parseString = opts.input;
+
+    startProduction = {
+        'address': address,
+        'address-list': addressList,
+        'angle-addr': angleAddr,
+        'from': fromSpec,
+        'group': group,
+        'mailbox': mailbox,
+        'mailbox-list': mailboxList,
+        'reply-to': replyToSpec,
+        'sender': senderSpec,
+    }[opts.startAt] || addressList;
+
+    if (!opts.strict) {
+        initialize();
+        opts.strict = true;
+        parsed = startProduction(parseString);
+        if (opts.partial || !inStr()) {
+            return giveResult(parsed);
+        }
+        opts.strict = false;
+    }
+
+    initialize();
+    parsed = startProduction(parseString);
+    if (!opts.partial && inStr()) { return null; }
+    return giveResult(parsed);
+}
+
+function parseOneAddressSimple(opts) {
+    return parse5322(handleOpts(opts, {
+        oneResult: true,
+        rfc6532: true,
+        simple: true,
+        startAt: 'address-list',
+    }));
+}
+
+function parseAddressListSimple(opts) {
+    return parse5322(handleOpts(opts, {
+        rfc6532: true,
+        simple: true,
+        startAt: 'address-list',
+    }));
+}
+
+function parseFromSimple(opts) {
+    return parse5322(handleOpts(opts, {
+        rfc6532: true,
+        simple: true,
+        startAt: 'from',
+    }));
+}
+
+function parseSenderSimple(opts) {
+    return parse5322(handleOpts(opts, {
+        oneResult: true,
+        rfc6532: true,
+        simple: true,
+        startAt: 'sender',
+    }));
+}
+
+function parseReplyToSimple(opts) {
+    return parse5322(handleOpts(opts, {
+        rfc6532: true,
+        simple: true,
+        startAt: 'reply-to',
+    }));
+}
+
+function handleOpts(opts, defs) {
+    function isString(str) {
+        return Object.prototype.toString.call(str) === '[object String]';
+    }
+
+    function isObject(o) {
+        return o === Object(o);
+    }
+
+    function isNullUndef(o) {
+        return o === null || o === undefined;
+    }
+
+    var defaults, o;
+
+    if (isString(opts)) {
+        opts = { input: opts };
+    } else if (!isObject(opts)) {
+        return null;
+    }
+
+    if (!isString(opts.input)) { return null; }
+    if (!defs) { return null; }
+
+    defaults = {
+        oneResult: false,
+        partial: false,
+        rejectTLD: false,
+        rfc6532: false,
+        simple: false,
+        startAt: 'address-list',
+        strict: false,
+        atInDisplayName: false,
+        commaInDisplayName: false,
+        addressListSeparator: ',',
+    };
+
+    for (o in defaults) {
+        if (isNullUndef(opts[o])) {
+            opts[o] = !isNullUndef(defs[o]) ? defs[o] : defaults[o];
+        }
+    }
+    return opts;
+}
+
+parse5322.parseOneAddress = parseOneAddressSimple;
+parse5322.parseAddressList = parseAddressListSimple;
+parse5322.parseFrom = parseFromSimple;
+parse5322.parseSender = parseSenderSimple;
+parse5322.parseReplyTo = parseReplyToSimple;
```

Comment:
This looks like a valuable tool, but I'm not a fan of forking it here. We should probably make this addon an npm package so that its dependency can be pulled with standard tools.

Reply:
If you have the bandwidth to do that, this is of course doable. I did not want to force you to having to build your add-on using a complex toolchain.